src/jit/emitxarch.cpp

```diff
@@ -492,10 +492,13 @@ emitter::code_t emitter::AddRexWPrefix(instruction ins, code_t code)
     if (UseVEXEncoding() && IsAVXInstruction(ins))
     {
         // W-bit is available only in 3-byte VEX prefix that starts with byte C4.
-        assert(hasVexPrefix(code));
+        if (hasVexPrefix(code))
```

Comment:
I also thought about doing:
```
if (TakesVexPrefix(ins))
{
    assert(hasVexPrefix(code));
}
```

Which may be better. Thoughts?

Reply:
I think it might be marginally easier to understand, but I don't feel strongly.

Reply:
I was also thinking that it might be more robust (for the Release configuration) in the case that the assertion was incorrect.

That is, the VEX prefix being there is the default for SSE/AVX instructions, and the `AddVexPrefix` code just `Or`s the prefix in. So, if (somehow) there was some release code for which the initial prefix was missed but for which it was added in later (we have the `AddVexPrefixIfNeededAndNotPresent` in a few places) the code would still generally work.

Reply:
I see your point.

Reply:
I convinced myself that the other way is both better and more readable, so I've updated to it.